src/ts/features/block-navigation-mode/blockNavigation.ts

```diff
@@ -0,0 +1,169 @@
+import {assumeExists} from '../../utils/assert'
+import {Selectors} from '../../roam/roam-selectors'
+
+export type Panel = 'MAIN' | 'SIDEBAR'
+
+type State = {
+    mainBlockId: string | null
+    sideBlockId: string | null
+    panel: Panel
+}
+
+export const state: State = {
```

Comment:
I believe this should be a singleton object that'd encapsulate management of the state vs allowing this to be globally writable